src/ProjectAudioManager.cpp

```diff
@@ -622,33 +623,17 @@ bool ProjectAudioManager::DoRecord(AudacityProject &project,
             if (prerollTrack)
                   transportTracks.prerollTracks.push_back(wt);
 
-            // A function that copies all the non-sample data between
-            // wave tracks; in case the track recorded to changes scale
-            // type (for instance), during the recording.
-            auto updater = [](Track &d, const Track &s){
-               auto &dst = static_cast<WaveTrack&>(d);
-               auto &src = static_cast<const WaveTrack&>(s);
-               dst.Reinit(src);
-            };
-
```

Comment:
Do you understand the consequences of deleting this?

I'll tell you:  If you make certain appearance changes to a track that you are recording to, those changes should be remembered in undo/redo history, but sequenced in that history before the recording, when the recording finishes.

I don't see why deleting these lines has any relation to the issue description.

Reply:
Well, the problem i was trying to solve here is when recording starts to an empty track but not from the beginning, Reinit also updates an offset of the new clip of the pending track so that it becomes 0 (i.e. recording starts from the beginning), but i supposed that this call may be redundant because all settings are copied when pending track is created, but missed that updater could be also called during recording (and seems like this changes aren't remembered when not recording). What the purpose of the updating the offset of the dst track? I see a two options here: to remove SetOffset from Reinit, or save initial offset before calling Reinit and restore it after the call (which is propably a workaround).

Reply:
Rather than deleting the code entirely, could the behavior of Reinit be fixed?

Reply:
removing 
`this->SetOffset(orig.GetOffset());`
from Reinit will solve the problem with recording to a empty track, but i do not understand what is the use of this line since Audacity does not allow clip moving during selection?